packages/react-native/ReactCommon/react/renderer/graphics/Transform.cpp

```diff
@@ -144,24 +148,33 @@ Transform Transform::Rotate(Float x, Float y, Float z) {
 }
 
 Transform Transform::FromTransformOperation(
-    TransformOperation transformOperation) {
+    TransformOperation transformOperation, Float viewWidth, Float viewHeight) {
   if (transformOperation.type == TransformOperationType::Perspective) {
-    return Transform::Perspective(transformOperation.x);
+    return Transform::Perspective(transformOperation.x.value);
   }
   if (transformOperation.type == TransformOperationType::Scale) {
     return Transform::Scale(
-        transformOperation.x, transformOperation.y, transformOperation.z);
+        transformOperation.x.value, transformOperation.y.value, transformOperation.z.value);
```

Comment:
This is quite a type-unsafe API now, as it's very easy to "forget" to check whether it's a percentage of a point value. Yoga used to require you to call something like `resolveValue`, can we make something explicit?

Reply:
Updated